### PR TITLE
chore: extend the code documentation

### PR DESCRIPTION
**Pull request by andreas-kupries**
Extend the documentation we have for our code, in the code's comments.
Look into generating documentation from these comments.

### CODE REVIEW

File: internal/api/v1/applications.go [hunk 1]
```diff
@@ -230,8 +235,9 @@ func (hc ApplicationsController) ServiceApps(w http.ResponseWriter, r *http.Requ
 	return nil
 }
 
-// Update modifies the specified application. currently this is only
-// the number of instances to run.
+// Update handles the API endpoint /orgs/:org/applications/:app (POST)
```
Comment (rohitsakala) [inline internal/api/v1/applications.go:238]: Is Update a PATCH operation ?

File: internal/api/v1/applications.go [hunk 2]
```diff
@@ -75,7 +77,8 @@ func (hc ApplicationsController) Create(w http.ResponseWriter, r *http.Request)
 	return nil
 }
 
-// Index lists all the known applications, with and without workload.
```
Comment (rohitsakala) [inline internal/api/v1/applications.go:78]: Some comments have operations like GET POST mentioned but some are not.

File: internal/api/v1/models/app.go [hunk 3]
```diff
@@ -30,18 +30,25 @@ func (a *App) AppRef() AppRef {
 	return NewAppRef(a.Name, a.Organization)
 }
 
+// AppList is just a collection of app references
```
Comment (rohitsakala) [inline internal/api/v1/models/app.go:33]: AppList is a collection is better ?

File: internal/auth/certs.go [hunk 4]
```diff
@@ -1,5 +1,3 @@
-// Package auth collects structures and functions around the
```
Comment (rohitsakala) [inline internal/auth/certs.go:1]: Is this not required ?

**Review by rohitsakala — CHANGES_REQUESTED** [review]

File: internal/api/v1/applications.go [hunk 5]
```diff
@@ -23,11 +23,13 @@ import (
...
 )
 
+// ApplicationsController represents all functionality of the API related to applications
 type ApplicationsController struct {
 	conn *websocket.Conn
 }
 
-// Create creates a new and empty application. I.e. without a workload.
+// Create handles the API endpoint /orgs/:org/applications (POST)
+// It creates a new and empty application. I.e. without a workload.
```
[the remark this answers is not in the document]
Comment (andreas-kupries) [reply internal/api/v1/applications.go:32]: I started a new sentence.

File: internal/api/v1/applications.go (hunk 1 above)
Comment (andreas-kupries) [reply internal/api/v1/applications.go:238]: Huh. Yes.
```suggestion
// Update handles the API endpoint /orgs/:org/applications/:app (PATCH)
```

File: internal/api/v1/applications.go (hunk 2 above)
Comment (andreas-kupries) [reply internal/api/v1/applications.go:78]: Yes. Added the information only when I noticed same url, different http methods. Can be consolidated to always have the method.

File: internal/api/v1/models/app.go (hunk 3 above)
Comment (andreas-kupries) [reply internal/api/v1/models/app.go:33]:
```suggestion
// AppList is a collection of app references
```

File: internal/auth/certs.go (hunk 4 above)
Comment (andreas-kupries) [reply internal/auth/certs.go:1]: No. The description is at the top of `auth.go`. Repeating it here causes the docs generator to repeat it as well in the output.

**Pull request merged**